airflow/providers/amazon/aws/executors/ecs/Setup_guide.md

```diff
@@ -0,0 +1,148 @@
+
+In order to create a Task Definition for the ECS Cluster that will work with Apache Airflow, you will need a Docker image that is properly configured. See the [Dockerfile](README.md#dockerfile-for-ecs-executor) section for instructions on how to do that.
+
+Once the image is built, it needs to be put in a repository where it can be pulled by ECS. There are multiple ways to accomplish this. This guide will go over doing this using Amazon Elastic Container Registry (ECR).
+
+### Create an ECR Repository
+
+1. Log in to your AWS Management Console and navigate to the ECR service.
+2. Click Create repository.
+3. Name the repository and fill out other information as required.
+4. Click Create Repository.
+5. Once the repository has been created, click on the repository. Click on the "View push commands" button on the top right.
+6. Follow the instructions to push the Docker image, replacing image names as appropriate. Ensure the image is uploaded by refreshing the page once the image is pushed.
+
+### Create ECS Cluster
+
+1. Log in to your AWS Management Console and navigate to the Amazon Elastic Container Service.
+2. Click "Clusters" then click "Create Cluster".
+3. Make sure that AWS Fargate (Serverless) is selected under Infrastructure.
+4. Select other options as required and click Create to create the cluster.
+
+### Create Task Definition
+
+1. Click on Task Definitions on the left hand bar, and click Create new task definition.
+2. Choose the Task Definition Family name. Select AWS Fargate for the Launch Type.
+3. Select or create the Task Role and Task Execution Role, and ensure the roles have the required permissions to accomplish their respective tasks. You can choose to create a new Task Execution role that will have the basic minimum permissions in order for the task to run.
+4. Select a name for the container, and use the image URI of the image that was pushed in the previous section. Make sure the role being used has the required permissions to pull the image.
+5. Add the following environment variables to the container:
+
+ - `AIRFLOW__DATABASE__SQL_ALCHEMY_CONN`, with the value being the PostgreSQL connection string in the following format using the values set during the [Database section](#create-the-rds-db-instance) above:
+
+```
+postgresql+psycopg2://<username>:<password>@<endpoint>/<database_name>
+```
+
+ - `AIRFLOW__ECS_EXECUTOR__SECURITY_GROUPS`, with the value being a comma separated list of security group IDs associated with the VPC used for the RDS instance.
+ - `AIRFLOW__ECS_EXECUTOR__SUBNETS`, with the value being a comma separated list of subnet IDs of the subnets associated with the RDS instance.
+
+6. Add other configuration as necessary for Airflow generally ([see here](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html)), the ECS executor ([see here](README.md#config-options)) or for remote logging ([see here](README.md#logging)).
```

Comment:
Yeah, this is the section that needs to be revamped to make sure config is consistent everywhere!

Reply:
Agreed, it's worth adding a note or a final sentence calling out that it's a best practice to keep your config the same in all locations.

CC @syedahsn

Reply:
> best practice

Not just best practice. Required/expected.